ratis-server/src/main/java/org/apache/ratis/server/impl/ConfigurationManager.java

```diff
@@ -61,10 +61,10 @@ private void setCurrentConf(RaftConfigurationImpl currentConf) {
     }
   }
 
-  synchronized void addConfiguration(RaftConfiguration conf) {
+  synchronized void addConfiguration(RaftConfiguration conf, long commitIndex) {
     final long logIndex = conf.getLogEntryIndex();
     final RaftConfiguration found = configurations.get(logIndex);
-    if (found != null) {
+    if (found != null && logIndex <= commitIndex) {
```

Comment:
Why changing this?  We should not allow two confs with the same index.

Reply:
> 7. B is elected as the new leader. The first thing B comes to power is to replicate a group configuration log L1 (old=[A,B,C], new=[A,B,C]) to establish its leader authority.

@SzyWilliam , I see the problem now.   This is similar to RATIS-2011.  We should truncate the confs when truncating the log.